riff-cli/cmd/delete.go

```diff
@@ -50,159 +46,86 @@ func Delete() (*cobra.Command, *DeleteOptions) {
 		Example: `  riff delete -n square
     or
   riff delete -f function/square`,
-		Args: utils.AliasFlagToSoleArg("filepath"),
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return doDelete(cmd, deleteOptions)
+			// From this point on, errors are execution errors, not misuse of flags
+			cmd.SilenceUsage = true
+			actingClient := realKubeCtl
+			if deleteOptions.DryRun {
+				actingClient = dryRunKubeCtl
+			}
+			return deleteFunctionByName(deleteOptions, actingClient, realKubeCtl)
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 
 			deleteOptions.Namespace = utils.GetStringValueWithOverride("namespace", *cmd.Flags())
-
-			// If name and no file path given, skip this step
-			if deleteOptions.FilePath != "" && deleteOptions.FunctionName == "" {
-				return  options.ValidateNamePathOptions(&deleteOptions.FunctionName, &deleteOptions.FilePath)
+			if deleteOptions.FunctionName == "" {
+				cwd, err := os.Getwd()
+				if err != nil {
+					return err
+				}
+				if deleteOptions.FunctionName, err = functions.FunctionNameFromPath(cwd); err != nil {
+					return err
+				}
 			}
 			return nil
 		},
 	}
 
 	deleteCmd.Flags().BoolVar(&deleteOptions.All, "all", false, "delete all resources including topics, not just the function resource")
 	deleteCmd.Flags().BoolVar(&deleteOptions.DryRun, "dry-run", false, "print generated function artifacts content to stdout only")
-	deleteCmd.Flags().StringVarP(&deleteOptions.FilePath, "filepath", "f", "", "path or directory used for the function resources (defaults to the current directory)")
 	deleteCmd.Flags().StringVarP(&deleteOptions.FunctionName, "name", "n", "", "the name of the function (defaults to the name of the current directory)")
 	deleteCmd.Flags().StringVar(&deleteOptions.Namespace, "namespace", "", "the namespace used for the deployed resources (defaults to kubectl's default)")
 
 	return deleteCmd, &deleteOptions
 }
 
-func doDelete(cmd *cobra.Command, opts DeleteOptions) error {
-
-	var cmdArgs []string
-	var message string
-
-	if opts.FilePath == "" && opts.FunctionName != "" {
-		err := deleteFunctionByName(opts)
-		if err != nil {
-			cmd.SilenceUsage = true
-		}
-		return err
-	}
-
-	var err error
-	opts.FunctionName, err = functions.FunctionNameFromPath(opts.FilePath)
-	if err != nil {
-		cmd.SilenceUsage = true
-		return err
+func deleteFunctionByName(opts DeleteOptions, actingClient kubectl.KubeCtl, queryClient kubectl.KubeCtl) error {
+	cmdArgs := []string{"delete"}
+	cmdArgs = append(cmdArgs, "function", opts.FunctionName)
+	if opts.Namespace != "" {
+		cmdArgs = append(cmdArgs, "--namespace", opts.Namespace)
 	}
 
-	abs, err := functions.AbsPath(opts.FilePath)
-	if err != nil {
-		cmd.SilenceUsage = true
+	if err := deleteResources(cmdArgs, actingClient); err != nil {
```

Comment:
we can't delete the function yet if `--all` is specified since we need the function resource to look up the topics

Reply:
Doh! TDD for the win!